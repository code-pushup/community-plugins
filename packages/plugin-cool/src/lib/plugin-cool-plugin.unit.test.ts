import { describe, expect, it } from 'vitest';
import { pluginConfigSchema } from '@code-pushup/models';
import pluginCoolPlugin from './plugin-cool-plugin';
import {omitVariablePluginData} from "testing-utils";

describe('pluginCoolPlugin', () => {
  it('should create valid plugin-cool plugin configuration', async () => {
    const pluginCoolPluginConfig = pluginCoolPlugin({});
    expect(() =>
      pluginConfigSchema.parse(pluginCoolPluginConfig),
    ).not.toThrow();
    expect(omitVariablePluginData(pluginCoolPluginConfig)).toMatchSnapshot();
  });
});
