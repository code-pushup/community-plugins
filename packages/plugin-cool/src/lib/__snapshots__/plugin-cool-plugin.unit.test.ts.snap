// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`pluginCoolPlugin > should create valid plugin-cool plugin configuration 1`] = `
{
  "audits": [
    {
      "description": "plugin-cool audit ",
      "slug": "plugin-cool-audit",
      "title": "plugin-cool Audit",
    },
  ],
  "description": "plugin-cool plugin for Code PushUp",
  "icon": "folder",
  "runner": [Function],
  "slug": "plugin-cool",
  "title": "plugin-cool",
}
`;
